tutorials/foreign-object.html

```diff
@@ -0,0 +1,813 @@
+    -webkit-user-select: text;  /* prefix needed for Safari */
+    user-select: text;
+}
+
+g[data-type="example.ForeignObject"] p {
+    cursor: auto;
+}
+</code></pre>
+
+        <p>
+            As our <code>foreignObject</code> might contain lots of HTML, there are moments when we certainly want to
+            be more concise when creating custom elements. Taking advantage of
+            <a href="/docs/jointjs#util.svg" target="_blank"><code>util.svg</code></a> in JointJS,
+            we can convert a string into a <a href="/docs/jointjs#dia.Cell.markup" target="_blank""><code>markup</code></a>
+            object while declaring a <code>Cell</code>. The resulting custom element will be more compact, and maybe easier to
+            read for some users.
+        </p>
+
+        <pre><code>const Card = joint.dia.Element.define('example.ForeignObject', {
+    attrs: {
+        body: {
+            width: 'calc(w)',
+            height: 'calc(h)',
+            fill: {
+                type: 'linearGradient',
+                stops: [
+                    { offset: 0, color: '#ff5c69' },
+                    { offset: 0.5, color: '#ff4252' },
+                    { offset: 1, color: '#ed2637' }
+                ]
+            }
+        },
+        foreignObject: {
+            width: 'calc(w-12)',
+            height: 'calc(h-12)',
+            x: 6,
+            y: 6
+        }
+    },
+}, {
+    // The /* xml */ comment is optional.
+    // It is used to tell the IDE that the markup is XML.
+    markup: joint.util.svg/* xml */`
+        &lt;style&gt;
+            div.card__background {
+                background-color: #131e29;
+                height: 100%;
+            }
+            p.card__text {
+                color: #F2F2F2;
+                font: 16px sans-serif;
+                padding: 10px;
+                margin: 0;
+            }             
+        &lt;/style&gt;
+        &lt;rect @selector="body"/&gt;
+        &lt;foreignObject @selector="foreignObject"&gt;
+            &lt;div @selector="background"
+                xmlns="http://www.w3.org/1999/xhtml"
+                class="card__background"
+            &gt;
+                &lt;p @selector="text"
+                    class="card__text"
+                &gt;
+                    Lorem ipsum dolor sit amet, consectetur adipiscing elit.
+                &lt;/p&gt;
+            &lt;/div&gt;
+        &lt;/foreignObject&gt;
+    `
+});
+</code></pre>
+
+        <h3>A more interactive example</h3>
+
+        <p>
+            One typical method which diagram libraries utilize to allow users work with HTML is to render HTML on top of an underlying element.
+            While this approach can be successful, it comes with a lot of additional complexity such as the need to keep dimensions and position 
+            of the HTML in sync with the element itself which isn't ideal. It's possible to use this approach with JointJS, and still to this day,
```

Comment:
Remove "which isn't ideal"... it's already clear that it is not ideal because of the part of the sentence which says that the approach comes with a lot of additional complexity